lightningd/jsonrpc.c

```diff
@@ -611,8 +611,22 @@ struct rpc_command_hook_payload {
 static void rpc_command_hook_serialize(struct rpc_command_hook_payload *p,
                                        struct json_stream *s)
 {
+	const jsmntok_t *tok;
+	size_t i;
+	char *key;
 	json_object_start(s, "rpc_command");
-	json_add_tok(s, "rpc_command", p->request, p->buffer);
+
+#ifdef COMPAT_V081
+	if (deprecated_apis)
+		json_add_tok(s, "rpc_command", p->request, p->buffer);
```

Comment:
heh i came across this yesterday. can you remove the outer object and instead do this line? it should achieve the same result without needing to iterate through the object.

Reply:
I tried to do that, however if I use `json_add_tok` to fill in `{"rpc_command": <copy>}` I cannot re-enter the `rpc_command` key to then add the compat version, i.e., `{"rpc_command": {"rpc_command": <copy>, <copied-fields}}`. Once we remove the `COMPAT_V081` flag we can actually do this and avoid iterating through, but until then I'm afraid we need to keep this.

Reply:
ack. i understand what this outputs now. thanks.